### PR TITLE
DOC: use footcite in mne/datasets/sleep_physionet/age.py

### LINKED ISSUE #9005: clean up XDAWN example
in `examples/preprocessing/plot_xdawn_denoising.py` the references are printed at the top; move them to the bottom like they are in most other examples/tutorials.

While you're in there, read through the example and make sure you understand it; improve the wording/explanation of any parts that are awkward or unclear. How-to examples should be fairly short, but if adding a little something (a few extra sentences, one more plot, etc) would make it more informative or clear, add it.

### PR DESCRIPTION
**Pull request by catalinamagalvan**
#### Reference issue
Fixes #9105.

#### What does this implement/fix?
Convert references in mne/datasets/sleep_physionet/age.py to use footcite / footbibliography

### CODE REVIEW

**Comment by drammock** [comment]
@catalinamagalvan there are several "trailing whitespace" errors (check the "files changed" tab of this pull request). Can you fix those?

**Review by sappelhoff — APPROVED** [review]
lgtm in principle, however you might get a CI failure due to line 24 being very long (I don't know for sure, we'll see).

It's usually good to insert line breaks to keep blocks of text at approximately the same "block size"

File: mne/datasets/sleep_physionet/age.py [hunk 1]
```diff
@@ -20,9 +20,8 @@ def fetch_data(subjects, recording=(1, 2), path=None, force_update=False,
                update_path=None, base_url=BASE_URL, on_missing='raise',
                verbose=None):  # noqa: D301
     """Get paths to local copies of PhysioNet Polysomnography dataset files.
-
```
Comment (drammock) [inline mne/datasets/sleep_physionet/age.py:23]: docstring error:
```
WARNING: [numpydoc] Validation warnings while processing docstring for 'mne.datasets.sleep_physionet.age.fetch_data':
  SS06: Summary should fit in a single line
```

The blank line here is important, it needs to be put back.  Same error, different test:
https://dev.azure.com/mne-tools/mne-python/_build/results?buildId=12350&view=logs&j=3e7b9abc-cf32-5e9b-ccb3-9355b11c347c&t=9989323c-d330-5eb0-bad7-a6f163ea363a&l=16

File: mne/datasets/sleep_physionet/age.py [hunk 2]
```diff
@@ -20,9 +20,8 @@ def fetch_data(subjects, recording=(1, 2), path=None, force_update=False,
...
     This will fetch data from the publicly available subjects from PhysioNet's
-    study of age effects on sleep in healthy subjects [1]_[2]_. This
+    study of age effects on sleep in healthy subjects :footcite:`MourtazaevEtAl1995,GoldbergerEtAl2000`. This
```
Comment (drammock) [inline mne/datasets/sleep_physionet/age.py:24]: this won't actually cause an error because we already have a `# noqa E501` comment at the end of this docstring.  But we still should constrain the line to fit:
```suggestion
    study of age effects on sleep in healthy subjects
    :footcite:`MourtazaevEtAl1995,GoldbergerEtAl2000`. This
```

File: tutorials/source-modeling/plot_dipole_orientations.py [hunk 3]
```diff
@@ -118,6 +118,7 @@
 _, time_max = stc.get_peak(hemi='lh')
 brain_fixed = stc.plot(surface='white', subjects_dir=subjects_dir,
                        initial_time=time_max, time_unit='s', size=(600, 400))
+mne.viz.set_3d_view(figure=fig, azimuth=180, distance=0.25)
```
Comment (britta-wstnr) [inline tutorials/source-modeling/plot_dipole_orientations.py:121]: This change looks unrelated to me, can you check? :-)

Comment (catalinamagalvan) [reply tutorials/source-modeling/plot_dipole_orientations.py:121]: Sure! My mistake! Thanks

**Review by sappelhoff — APPROVED** [review]
thanks for the fixes. I slightly changed your PR title.

**Review by drammock — APPROVED** [review]

**Comment by drammock** [comment]
thanks @catalinamagalvan!

**Comment by catalinamagalvan** [comment]
Thanks @britta-wstnr, @drammock and @sappelhoff :)
